num_enum_derive/src/lib.rs

```diff
@@ -62,6 +62,63 @@ pub fn derive_into_primitive(input: TokenStream) -> TokenStream {
     })
 }
 
+/// Generates a `const_into(self)` method which can be used to extract the primitive value from the
+/// enum in `const` contexts.
+///
+/// ## Allows turning an enum into a primitive.
+///
+/// ```rust
+/// use num_enum::ConstIntoPrimitive;
+///
+/// #[derive(ConstIntoPrimitive)]
+/// #[repr(u8)]
+/// enum Number {
+///     Zero,
+///     One,
+/// }
+///
+/// const zero: u8 = Number::Zero.const_into();
+/// assert_eq!(zero, 0u8);
+/// ```
+#[proc_macro_derive(ConstIntoPrimitive, attributes(num_enum, catch_all))]
+pub fn derive_const_into_primitive(input: TokenStream) -> TokenStream {
```

Comment:
I'm a little concerned that this implementation and the `derive_into_primitive` may drift in the future - can you extract a common function that both this and `derive_into_primitive` can call, which is parameterised on a boolean/enum, to avoid that drift? (Same for the other derives)

Reply:
This part was hammered out by rote to get a prototype working and I will absolutely see if I can join them together as you say!